lua/starfall/libs_sh/vectors.lua

```diff
@@ -388,26 +388,39 @@ function vec_methods:normalize()
 	self[3] = self[3] / len
 end
 
+local math_sin, math_cos = math.sin, math.cos
+local deg2rad = math.pi/180
+
 --- Rotate the vector by Angle b.
 -- Self-Modifies. Does not return anything
 -- @param Angle b Angle to rotate by.
 function vec_methods:rotate(b)
-	local vec = unwrap(self)
-	vec:Rotate(aunwrap(b))
+	checktype(b, ang_meta)
+	local y, p, r = b[1] * deg2rad, b[2] * deg2rad, b[3] * deg2rad
```

Comment:
This is no longer correct. The order for angle components is: `pitch, yaw, roll`.
Keep the indices in this order, but swap variables around to `p, y, r`:
```lua
local p, y, r = b[1] * deg2rad, b[2] * deg2rad, b[3] * deg2rad
```